[WIP] Chapter 7 revision (#443)
worldbank/d4di: chapters/7-publication.tex

% what exactly is a dynamic document
The most widely used software
for dynamically managing academic manuscripts is {\LaTeX} (pronounced ``lah-tek'').
\index{\LaTeX}
{\LaTeX} is a document preparation and typesetting system with a unique syntax.
While this tool has a significant learning curve,
its enormous flexibility in terms of operation, collaboration, output formatting, and styling
make it the primary choice for most large technical outputs.
It enables a streamlined collaboration workflow
that allows explicitly encoded references to raw output files.
This means that, whenever raw outputs are updated and the document is compiled,
it will automatically include the latest versions of all outputs.
This is not possible by default in Microsoft Word,
although there are add-ons that produce similar functionality.
In Word, you have to copy and paste each object
whenever tables, graphs, or other inputs are updated.
This creates complex inefficiency: updates may be accidentally excluded,
and ensuring they are not becomes more difficult as the document grows.
As time goes on, it becomes more and more likely
that a mistake will be made or something will be missed.

> This is a broadly unsuitable way to prepare technical documents.

Author: Done in 7177b7d9dfad7ecfadb5b2f1db534324ffeac5f7

though different services vary the number of collaborators and documents allowed at each tier.
Fourth, and most usefully, some implementations provide a ``rich text'' editor
that behaves pretty similarly to familiar tools like Word,
so that collaborators can write text directly into the document without worrying too much
about the underlying {\LaTeX} coding.
Cloud services also usually offer a convenient selection of templates
so it is easy to start up a project and see results right away
without needing to know a lot of the code that controls document formatting.

% Disadvantages of cloud-based implementations
Cloud-based implementations of {\LaTeX} also have disadvantages.
There is still some up-front learning required, unless you're using the rich text editor.
Continuous access to the Internet is necessary,
and updating figures and tables requires a bulk file upload that is tough to automate.
They vary dramatically in their ability to integrate
with file systems where you store your code and outputs,
and so you will need to practice an integrated workflow depending what is available to you.
Despite this, we believe that with minimal learning and workflow adjustments,

> cloud-based implementations are often the easiest way to allow coauthors to write and edit in \LaTeX,

Author: ```suggestion
cloud-based implementations are often the easiest way to allow coauthors to write and edit in {\LaTeX},
```

While software-specific datasets are acceptable accompaniments to the code
(since those precise materials are probably necessary),
you should also consider releasing datasets in plain text formats
such as CSV files with accompanying codebooks,
since these can be used by any researcher.
Additionally, you should also release PDF or code versions of
the data collection instrument or survey questionnaire
so that readers can understand which data components are
collected directly in the field and which are derived.
With your analytical dataset,
you should also release the code
that constructs any derived measures
from the raw dataset,
so that others can learn from your work and adapt it as they like.

\section{Preparing a complete reproducibility package}

Major journals now often require that,
in addition to the data required to obtain your results,
you provide the code that exactly recreates those results.
Some even require being able to reproduce the results themselves
before they will approve a paper for publication.\sidenote{
  \url{https://www.aeaweb.org/journals/policies/data-code}}
This set of materials, taken together,

> is often referred to as a ``reproducibility package''.

Author: ```suggestion
is often referred to as a \textbf{reproducibility package'}.
```

Make sure system settings like \texttt{version}, \texttt{matsize}, and \texttt{varabbrev} are set.

% tracking code and outputs
Finally, make sure that code inputs and outputs are clearly identified.
A new user should, for example, be able to easily find and remove
any files created by the code so that they can be recreated quickly.
They should also be able to quickly map all the outputs of the code
to the locations where they are placed in the associated published material,
so ensure that the raw components of figures or tables are clearly identified.
Documentation in the master script is often used to indicate this information.
For example, outputs should clearly correspond by name to an exhibit in the paper, and vice versa.
(Supplying a compiling {\LaTeX} document can support this.)
Code and outputs which are not used should be removed before publication.

\subsection{Releasing a reproducibility package}

% finding a place to publish the package
Once your data and code are polished for public release,
all you need to do is find a place to publish your materials.
This is slightly easier said than done,
as there are a few variables to take into consideration

> and, at the time of writing, no global consensus on the best solution

Author: Reject change, this is standard language

You may be required to put your materials into the public domain;
if not, it is common to only require attribution and citation for their reuse,
without putting any barriers or restrictions to accessing the code.

% pros and cons of GitHub
One option for releasing a reproducibility package is GitHub.
Making a public GitHub repository is completely free.
It can hold any file types,
provide a structured, compressed download of your whole project,
and allow others to look at alternate versions or histories easily.
It is straightforward to simply upload a fixed directory to GitHub,
apply a sharing license, and obtain a URL for the whole package.
There is a strict size restriction of 100MB per file and
a restriction on the size of the repository as a whole,
so larger projects will need alternative solutions.
However, GitHub is not the ideal platform to release reproducibility packages.
It is built to version control code, and to facilitate collaboration on it.
It is not an archive, meaning that it does not guarantee the permanence
of uploaded materials, it does not guarantee the permanence of the access URL,
and it does not manage citations or non-code licenses by default.

> Features to look for in a platform to release reproducibility packages
> and that are not offered by GitHub, include:

Reviewer: Taking the GitHub mention out, as it could be argue that some of these features are available through GitHub

```suggestion
Features to look for in a platform to release reproducibility packages include:
```

Emerging technologies such as the ``containerization'' approach of CodeOcean\sidenote{
  \url{https://codeocean.com}}
offer to store both code and data in one repository,
and also provide an online workspace in which others can execute and modify your code
without having to download your tools and match your local environment
when packages and other underlying software may have changed since publication.
In a similar spirit, attempts to create ``metadata archives''
that link all the different materials, releases, and publications
relating to specific research projects or studies
offer an avenue for improving the organization of these materials in the future.

% pre-prints
In addition to code and data,
you may also want to release an author's copy or preprint
of the article itself along with these raw materials.
Check with your publisher before doing so;
not all journals will accept material that has been publicly released
before its formal publication date, although,
in most development research fields,
the release of working papers is a fairly common practice.
This can be done on a number of preprint websites,
many of which are topic-specific.\sidenote{

>   \url{https://en.wikipedia.org/wiki/ArXiv}}

Reviewer: ```suggestion
  \url{https://arxiv.org/}}
```
